src/Voron/Impl/FreeSpace/StreamBitArray.cs

```diff
@@ -110,6 +114,180 @@ public bool Get(int index)
             return (_inner[index >> 5] & (1 << (index & 31))) != 0;
         }
 
+        public int? GetContinuousRangeStart(int num)
+        {
+            switch (num)
```

Comment:
Split this into separate methods for the < 32 and > 32 case.
For better inlining

Reply:
There is a much better algorithm to do that. @grisha-kotler is going to create a microbenchmark with the different variants in order to be able to optimize it for the different architectures too.

Reply:
done